util/hookable/hookable.cpp

```diff
@@ -342,6 +404,115 @@ J9HookRegister(struct J9HookInterface **hookInterface, uintptr_t taggedEventNum,
 					(emptyRecord->next->agentID >= agentID)))
 		) {
 			emptyRecord->function = function;
+			emptyRecord->callsite = NULL;
```

Comment:
Rather than modifying both `J9HookRegister` and the new `J9HookRegisterWithCallSite`, would it be better to have the old function call the new one passing 'NULL' as the callsite?  

This would mean there aren't two implementations that need to be kept in sync.